dnfdaemon-client/commands/repolist/repolist.cpp

```diff
@@ -0,0 +1,189 @@
+
+    auto repolist = ctx.arg_parser.add_new_command(command);
+    repolist->set_short_description("display the configured software repositories");
+    repolist->set_description("");
+    repolist->set_named_args_help_header("Optional arguments:");
+    repolist->set_positional_args_help_header("Positional arguments:");
+    repolist->set_parse_hook_func([this, &ctx](
+                                      [[maybe_unused]] libdnf::cli::ArgumentParser::Argument * arg,
+                                      [[maybe_unused]] const char * option,
+                                      [[maybe_unused]] int argc,
+                                      [[maybe_unused]] const char * const argv[]) {
+        ctx.select_command(this);
+        return true;
+    });
+
+    repolist->register_named_arg(all);
+    repolist->register_named_arg(enabled);
+    repolist->register_named_arg(disabled);
+    repolist->register_positional_arg(repos);
+
+    ctx.arg_parser.get_root_command()->register_command(repolist);
+}
+
+class RepoDbus {
+public:
+    RepoDbus(dnfdaemon::KeyValueMap & rawdata) : rawdata(rawdata){};
+    std::string get_id() { return rawdata["id"]; }
+    std::string get_name() { return rawdata["name"]; }
+    bool is_enabled() { return rawdata["enabled"]; }
+    uint64_t get_size() { return rawdata["repo_size"]; }
+
+private:
+    dnfdaemon::KeyValueMap rawdata;
+};
+
+// repository list table columns
+enum { COL_REPO_ID, COL_REPO_NAME, COL_REPO_STATUS };
+
+static struct libscols_table * create_repolist_table(bool with_status) {
+    struct libscols_table * table = scols_new_table();
+    if (isatty(1)) {
+        scols_table_enable_colors(table, 1);
+        scols_table_enable_maxout(table, 1);
+    }
+    struct libscols_column * cl = scols_table_new_column(table, "repo id", 0.4, 0);
+    scols_column_set_cmpfunc(cl, scols_cmpstr_cells, NULL);
+    scols_table_new_column(table, "repo name", 0.5, SCOLS_FL_TRUNC);
+    if (with_status) {
+        scols_table_new_column(table, "status", 0.1, SCOLS_FL_RIGHT);
+    }
+    return table;
+}
+
+static void add_line_into_table(
+    struct libscols_table * table, bool with_status, const char * id, const char * descr, bool enabled) {
+    struct libscols_line * ln = scols_table_new_line(table, NULL);
+    scols_line_set_data(ln, COL_REPO_ID, id);
+    scols_line_set_data(ln, COL_REPO_NAME, descr);
+    if (with_status) {
+        scols_line_set_data(ln, COL_REPO_STATUS, enabled ? "enabled" : "disabled");
+        struct libscols_cell * cl = scols_line_get_cell(ln, COL_REPO_STATUS);
+        scols_cell_set_color(cl, enabled ? "green" : "red");
+    }
+}
+
+void CmdRepolist::run(Context & ctx) {
+    // prepare options from command line arguments
+    dnfdaemon::KeyValueMap options = {};
+    options["enable_disable"] = enable_disable_option->get_value();
+    std::vector<std::string> patterns_to_show;
+    if (patterns_to_show_options->size() > 0) {
```

Comment:
for better readability: `size > 0` means is not empty
`if (!patterns_to_show_options->empty()) {`